src/session_base.cpp

```diff
@@ -434,13 +434,19 @@ void zmq::session_base_t::engine_error (
     switch (reason) {
         case stream_engine_t::timeout_error:
         case stream_engine_t::connection_error:
-            if (active)
+            if (active) {
                 reconnect ();
-            else
-                terminate ();
-            break;
+                break;
+            }
```

Comment:
this will trigger compiler warnings, so you'll want to add ```/* fallthrough */``` just before the next case

Reply:
actually the case above is already like that, so never mind

Reply:
I saw you changed that already, thanks :)